lisa/tools/kill.py

```diff
@@ -35,13 +36,20 @@ def by_name(self, process_name: str, signum: int = SIGKILL) -> None:
                 f"Kill for {process_name} did not find any processes to kill."
             )
 
-    def by_pid(self, pid: str, signum: int = SIGKILL) -> None:
-        self.run(
+    def by_pid(
+        self, pid: str, signum: int = SIGKILL, ignore_exit_code: bool = False
+    ) -> None:
+        result = self.run(
             f"-{signum} {pid}",
             shell=True,
             sudo=True,
             force_run=True,
-            expected_exit_code=0,
-            expected_exit_code_failure_message="fail to run "
-            f"{self.command} -{signum} {pid}",
         )
+
+        if result.exit_code != 0:
+            if ignore_exit_code and "No such process" in result.stdout:
```

Comment:
Remove the check of "No such process". Just use the ignore_exit_code result.

Reply:
The words and logic should be consistent. If it needs to ignore the "no such process" only, the variable name should be change like "ignore_not_exist".